src/core/sync/semaphore.d

```diff
@@ -50,6 +50,13 @@ else
     static assert(false, "Platform not supported");
 }
 
+/// Aliases the operating system-specific semaphore type.
+version(Windows)        alias SemaphoreHandle =  HANDLE;
+/// ditto
+else version(Darwin)    alias SemaphoreHandle = semaphore_t;
+/// ditto
+else version(Posix)     alias SemaphoreHandle = sem_t;
```

Comment:
I'll move this `alias` within the scope of the class, instead of prefixing it with the class name.

Reply:
done. Initially i just changed and move this because it was better for ddoc.